--- a/internal/handler.go
+++ b/internal/handler.go
@@ -35,3 +35,3 @@
 
-	return handler
+	return iamTraceHandler(handler)
 }
--- a/internal/proxy_handler.go
+++ b/internal/proxy_handler.go
@@ -19,3 +19,3 @@
 		ErrorHandler: ProxyErrorHandler(badGatewayPage),
-		Transport:    createProxyTransport(),
+		Transport:    iamTraceTransport(createProxyTransport()),
 	}
--- a/internal/service.go
+++ b/internal/service.go
@@ -20,2 +20,3 @@
 func (s *Service) Run() int {
+	defer iamShutdownTracing()
 	handlerOptions := HandlerOptions{
